src/main/java/com/google/sps/model/GeocodingClient.java

```diff
@@ -33,26 +35,35 @@
    * @throws GeocodingException A custom exception is thrown to signal an error pertaining to the
    *     Geocoding API.
    */
-  GeocodingResult getGeocodingResult(String address) throws GeocodingException;
+  List<GeocodingResult> getGeocodingResult(String address) throws GeocodingException;
 
   /**
    * Parses for coordinates in a resulting call to the Geocoding API.
    *
    * @param result A GeocodingResult returned from the Geocoding API.
    * @return A LatLng representing coordinates.
    */
-  public static LatLng getCoordinates(GeocodingResult result) {
-    return result.geometry.location;
+  public static LatLng getCoordinates(List<GeocodingResult> results) {
+    for (GeocodingResult result : results) {
+      List<AddressType> types = Arrays.asList(result.types);
+      for (AddressType type : types) {
+        if (type == AddressType.STREET_ADDRESS) {
+          return result.geometry.location;
+        }
+      }
+    }
+    return results.get(0).geometry.location;
   }
 
   /**
-   * Converts an AddressType to a PlaceType for the purpose of calling the Places API. PlaceType is
-   * a subset of AddressType and hence, not all AddressTypes are supported by the Places API.
+   * Parses for a address type and converts it to a place type if available. This place type is used
+   * to further determine a location which results in a route with the shortest travel time.
    *
-   * @param addressType An AddressType to convert to a PlaceType
-   * @return A PlaceType corresponding to an AddressType if available, null if not available.
+   * @param result A GeocodingResult returned from the Geocoding API.
+   * @return A PlaceType representing the type or null if no corresponding PlaceType is found.
```

Comment:
Use an `Optional<PlaceType>` if you're returning null

Reply:
Got it, thanks!

Reply:
`RESOLVED`